src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -417,5 +414,24 @@ private static string ToBase32StringSuitableForDirName(byte[] buff)
 
             return StringBuilderCache.GetStringAndRelease(sb);
         }
+        
+        // Note: This method returns path.Length for
+        // inputs like "/usr/foo/" on Unix. As such,
+        // it is NOT safe for being used to index
+        // the string without additional verification.
+        private static int FindFileNameIndex(string path)
```

Comment:
I'd recommend you put this as an internal in Common\src\System\IO\PathInternal.cs. You can then unit test it with the Common test project.